src/psyclone/psyir/nodes/acc_mixins.py

```diff
@@ -0,0 +1,131 @@
+classes.'''
+
+import abc
+
+from psyclone.psyir.nodes.reference import Reference
+
+
+class ACCAsyncMixin(metaclass=abc.ABCMeta):
+    '''
+    Class handling the common code to handle the async keyword on related acc
+    directives.
+
+    :param async_queue: Enable async support and attach it to the given queue.
+                        Can use False to disable, True to enable on default
+                        stream. Use int to attach to the given stream ID or
+                        use a variable Reference to say at runtime what stream
+                        to be used.
+    :type async_queue: bool | int | :py:class:`psyclone.core.Reference`
+    '''
+    def __init__(self, async_queue=False):
+        self.async_queue = async_queue
+
+    @property
+    def async_queue(self):
+        '''
+        :returns: whether or not async is enabled and if so, which queue this
+                  node is associated with. (True indicates the default stream.)
+                  Can use False to disable, True to enable on default stream.
+                  Int to attach to the given stream ID or use a variable
+                  Reference to say at runtime what stream to be used.
+        :rtype async_queue: bool | int | :py:class:`psyclone.core.Reference`
+        '''
+        return self._async_queue
+
+    @async_queue.setter
+    def async_queue(self, async_queue):
+        '''
+        :param async_queue: Enable async support and attach it to the given
+                            queue. Can use False to disable, True to enable on
+                            default stream. Int to attach to the given stream
+                            ID or use a variable Reference to say at runtime
+                            what stream to be used.
+        :type async_queue: bool | int | :py:class:`psyclone.core.Reference`
+        :raises TypeError: if `wait_queue` is of the wrong type
+        '''
+        # check
+        if (async_queue is not None and
+           not isinstance(async_queue, (bool, Reference, int))):
+            raise TypeError(f"Invalid async_queue value, expect Reference or "
+                            f"integer or None or bool, got : {async_queue}")
+
+        # assign
+        self._async_queue = async_queue
+
+    def _build_async_string(self):
+        '''
+        Build the async arg to concat to the acc directive when generating the
+        code.
+
+        :returns: The `async()` option to add to the directive.
+        :rtype: str
+        '''
+        # default
+        result = ""
+
+        # async
+        if self.async_queue is not None:
+            if isinstance(self.async_queue, bool):
+                if self.async_queue:
+                    result = " async()"
+            elif isinstance(self.async_queue, int):
+                result = f" async({self.async_queue})"
+            elif isinstance(self.async_queue, Reference):
+                from psyclone.psyir.backend.fortran import FortranWriter
```

Comment:
Please could you disable the pylint warning about this import by adding:

    # pylint: disable=import-outside-toplevel

on the line immediately before the import. Please could you also add:

    # TODO #1872 - this will be removed once a suitable Clause is implemented.